### PR TITLE
test/style/feat: Refactored several files:

### PR DESCRIPTION
**Pull request by ajago**
Hey, I am a developer moving from Java to Node right now, so looking to get a bit more experience of apps like this in Javascript. 
Am happy to split this into several PRs if needed, but thought it would be easier to handle it as one PR to start off with. I can create separate PRs for elements you are happy with if you dont want to take all the changes.

- Upload controller now processes upload in parallel
- Express config - uses es6 functions and optmised cors logic
- Secrets - all missing secrets are now logged out, minor tweak to mongoURI so it can be a constant
- Middleware - removed unnecessary brackets
- Integration tests - tidied up beforeEach and afterAll calls

### CODE REVIEW

**Comment by feredean** [comment]
Thank you so much for contributing to this project! I'll be back with feedback shortly.

File: src/config/express.ts [hunk 1]
```diff
@@ -14,10 +14,9 @@ export const setupExpress = (app: Express): void => {
     app.set("port", APP_PORT);
 
     const corsOptions = {
-        origin: function (origin: string, callback: Function): void {
+        origin: (origin: string, callback: Function): void => {
             if (!origin) return callback();
-            let match = false;
-            if (origin.match(new RegExp(CORS_REGEX))) match = true;
+            const match = origin.match(new RegExp(CORS_REGEX));
             callback(null, match);
```
Comment (feredean) [inline src/config/express.ts:20]: The callback takes as parameters `(err: Error | null, allow?: boolean)` see https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/cors/index.d.ts#L11, as such this won't work. You can test it by installing the frontend as well https://github.com/feredean/node-api-starter-angular-app

You can either go with:
```ts
const match = origin.match(new RegExp(CORS_REGEX)) ? true : false;
```
or
```ts
callback(null, !!match);
```
I'm fine with both

File: test/integration/integration.test.ts [hunk 2]
```diff
@@ -121,7 +121,7 @@ describe("API V1", (): void => {
                 await initMongo();
                 await registerValidUser(adminOpts);
             });
-            afterAll(async (): Promise<void> => disconnectMongo());
+            afterAll(disconnectMongo);
```
Comment (feredean) [inline test/integration/integration.test.ts:125]: Please also change line 73, 380 and 401 for consistency

File: src/middleware/index.ts [hunk 3]
```diff
@@ -20,7 +20,7 @@ export const isAuthenticated = async (req: Request, res: Response, next: NextFun
     try {
         if (!req.headers.authorization) return res.sendStatus(401);
         const token = req.headers.authorization.split("Bearer ")[1];
-        req.user = (await jwt.verify(token, SESSION_SECRET) as Express.User);
+        req.user = await jwt.verify(token, SESSION_SECRET) as Express.User;
```
Comment (feredean) [inline src/middleware/index.ts]: While looking into this change I noticed that the `jwt.verify` function is not async https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/jsonwebtoken/index.d.ts#L162. This means several changes are required. 
1. both `isAuthenticated` and `hasPermission` middlewares don't need to be async
2. as a result the middleware tests also no longer need to be async

I can update this in a future PR or if you want, you can commit these changes in this one

**Review by feredean — CHANGES_REQUESTED** [review]

File: src/config/express.ts (hunk 1 above)
Comment (ajago) [reply src/config/express.ts:20]: ah good point

File: test/integration/integration.test.ts (hunk 2 above)
Comment (ajago) [reply test/integration/integration.test.ts:125]: 👍

File: src/middleware/index.ts (hunk 3 above)
Comment (ajago) [reply src/middleware/index.ts]: I'll have a look at this tomorrow

**Comment by ajago** [comment]
addressed all the changes now

**Comment by feredean** [comment]
Looking good! One final thing, please rebase and squash everything into a single commit to keep the commit history clean

**Comment by ajago** [comment]
Was going to ask about your preference on that. Done.

**Review by feredean — APPROVED** [review]

**Comment by feredean** [comment]
Thank you very much for your contribution!

**Comment by feredean** [comment]
@ajago forgot to mention, if you're new to node you should take a look at these two projects
https://docs.nestjs.com/
https://typeorm.io/
I found them very interesting and they might become part of this project in the future

**Comment by ajago** [comment]
Thanks for the message, I've had a look at nestjs and it seems to be trying to introduce the same level of magic that you get in the Java Spring world, so will be interesting to see how it develops.

**Pull request merged**